cadence/contracts/NFTCatalogSnapshot.cdc

```diff
@@ -0,0 +1,31 @@
+// NFTCatalogSnapshot
+//
+// A snapshot of the NFT Catalog at a specific time.
+// This is provided in order to provide post-deprecation
+// support to the NFT Catalog `getCatalog` function.
+// which has been deprecated because it is nearing
+// execution limits when copied and used within a 
+// script or transaction.
+// https://github.com/dapperlabs/nft-catalog/issues/138
+// 
+
+pub contract NFTCatalogSnapshot {
+
+  access(self) var catalogSnapshot: {String : AnyStruct}?
+  pub var snapshotBlockHeight: UInt64?
+
+  access(account) fun setSnapshot(_ snapshot: {String : AnyStruct}) {
```

Comment:
Sure, this snapshot has a very singular purpose of storing the current state of the catalog so that it doesn't receive updates afterwards. We don't want people to make use of this snapshot much as we're deprecating the entry point to it, (the related getCatalog method), but this change is so that the result from the getCatalog method stops growing in size as the catalog grows.

With that in mind, there is no requirement of storing past or future snapshots. The plan is to only snapshot it once as well.